Tests/SourceKitLSPTests/MainFilesProviderTests.swift

```diff
@@ -19,78 +19,197 @@ import SourceKitLSP
 import XCTest
 
 final class MainFilesProviderTests: XCTestCase {
-
-  func testMainFilesChanged() async throws {
-    let ws = try await mutableSourceKitTibsTestWorkspace(name: "MainFiles")!
-    let indexDelegate = SourceKitIndexDelegate()
-    ws.tibsWorkspace.delegate = indexDelegate
-
-    final class TestMainFilesDelegate: MainFilesDelegate {
-      var expectation: XCTestExpectation
-      init(_ expectation: XCTestExpectation) { self.expectation = expectation }
-      func mainFilesChanged() {
-        expectation.fulfill()
-      }
-    }
-
-    let mainFilesDelegate = TestMainFilesDelegate(expectation(description: "main files changed"))
-    indexDelegate.registerMainFileChanged(mainFilesDelegate)
-
-    let a = ws.testLoc("a_func").docIdentifier.uri
-    let b = ws.testLoc("b_func").docIdentifier.uri
-    let c = ws.testLoc("c_func").docIdentifier.uri
-    let d = ws.testLoc("d_func").docIdentifier.uri
-    let unique_h = ws.testLoc("unique").docIdentifier.uri
-    let shared_h = ws.testLoc("shared").docIdentifier.uri
-    let bridging = ws.testLoc("bridging").docIdentifier.uri
-
-    XCTAssertEqual(ws.index.mainFilesContainingFile(a), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(b), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(c), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(d), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(unique_h), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(shared_h), [])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(bridging), [])
-
-    try ws.buildAndIndex()
-
-    XCTAssertEqual(ws.index.mainFilesContainingFile(a), [a])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(b), [b])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(c), [c])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(d), [d])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(unique_h), [d])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(shared_h), [c, d])
-    XCTAssertEqual(ws.index.mainFilesContainingFile(bridging), [c])
-
-    try await fulfillmentOfOrThrow([mainFilesDelegate.expectation])
-
-    try ws.edit { changes, _ in
-      changes.write(
-        """
-        #include "bridging.h"
-        void d_new(void) { bridging(); }
+  func testMainFileForHeaderInPackageTarget() async throws {
+    let ws = try await SwiftPMTestWorkspace(
+      files: [
+        "MyLibrary/include/MyLibrary.h": """
+        void bridging(void) {
+          int VARIABLE_NAME = 1;
+        }
+        """,
+        "MyLibrary/MyLibrary.c": """
+        #include "shared.h"
+        """,
+      ],
+      manifest: """
+        // swift-tools-version: 5.7
+
+        import PackageDescription
+
+        let package = Package(
+          name: "MyLibrary",
+          targets: [
+            .target(
+              name: "MyLibrary",
+              cSettings: [.define("VARIABLE_NAME", to: "fromMyLibrary"), .unsafeFlags(["-Wunused-variable"])]
```

Comment:
Took me a little bit to realize why we were doing this - worth a comment IMO (ie. that we're defining `VARIABLE_NAME` here to make sure that the build settings from the main file are being used for the header).

Reply:
I mentioned that we use `VARIABLE_NAME` to communicate the build settings in all test cases now.